f8a_worker/workers/init_package_flow.py

```diff
@@ -41,30 +39,44 @@ def get_upstream_entry(self, db, package, url):
             .order_by(desc(Upstream.updated_at))\
             .all()
 
-        # deactivate entries that have different upstream URL
         ret = None
         for entry in upstreams:
             if url is not None and entry.url != url:
                 self.log.info("Marking upstream entry with id '%s' and URL '%s' for deactivation, "
                               "substituting with upstream URL '%s'", entry.id, entry.url, url)
+                # deactivate entries that have different upstream URL
                 entry.deactivated_at = now
             else:
                 self.log.info("Reusing already existing and active upstream record with id '%d' "
                               "and upstream URL '%s'", entry.id, entry.url)
                 ret = entry
 
         if ret is None:
-            self.log.info("Creating new upstream record entry for package %s/%s and "
-                          "upstream URL '%s'", package.ecosystem.name, package.name, url)
-            ret = Upstream(
-                package_id=package.id,
-                url=url,
-                updated_at=None,
-                deactivated_at=None,
-                added_at=now,
-                active=True
-            )
-            db.add(ret)
+            older_upstream = db.query(Upstream) \
+                .filter(Upstream.package_id == package.id) \
+                .filter(Upstream.deactivated_at.isnot(None)) \
+                .filter(Upstream.url == url) \
+                .order_by(desc(Upstream.updated_at)) \
+                .first()
+            if older_upstream:
+                self.log.info("Activating older upstream record entry for package %s/%s and "
+                              "upstream URL '%s'", package.ecosystem.name, package.name, url)
+                older_upstream.updated_at = now
```

Comment:
Setting this here will break the following condition: https://github.com/jpopelka/fabric8-analytics-worker/blob/58a05e3d6e14c858acb5097cf83ee894f2f8af1f/f8a_worker/workers/init_package_flow.py#L100

Reply:
You're right. This line can be omitted since we set the `updated_at` later in `execute()`